src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1063,36 +1063,59 @@ internal async Task NegotiateEncryption(SslMode sslMode, NpgsqlTimeout timeout,
         {
             var password = Settings.SslPassword;
 
-            X509Certificate2? cert = null;
             if (!string.Equals(Path.GetExtension(certPath), ".pfx", StringComparison.OrdinalIgnoreCase))
             {
                 // It's PEM time
                 var keyPath = Settings.SslKey ?? PostgresEnvironment.SslKey ?? PostgresEnvironment.SslKeyDefault;
-                cert = string.IsNullOrEmpty(password)
-                    ? X509Certificate2.CreateFromPemFile(certPath, keyPath)
-                    : X509Certificate2.CreateFromEncryptedPemFile(certPath, password, keyPath);
+                if (string.IsNullOrEmpty(password) && string.IsNullOrEmpty(keyPath))
+                {
+                    // If there is no password or key provided, most likely there are multiple certificates
+                    // Where one is leaf and others are intermediate
+                    clientCertificates.ImportFromPemFile(certPath);
+                }
+                else
+                {
+                    var cert = string.IsNullOrEmpty(password)
+                        ? X509Certificate2.CreateFromPemFile(certPath, keyPath)
```

Comment:
So there's no way to read multiple certificates (as above with `X509Certificate2Collection.ImportFromPemFile()`) if the PEM file happens to be encrypted with a password or key? While with PKCS12 it's possible (below we do `X509CertificateLoader.LoadPkcs12CollectionFromFile`)?

Is this a weird/arbitrary .NET limitation, is it tracked somewhere?

Reply:
> There is no explicit API in .NET to load the first certificate from the chain with key/password and other with only public keys.

OK. I just want to confirm that the thing we're supporting here (loading first certificate with key/password, the rest of the certificates with public key only) is "standard" - or just supported by libpq (if libpq supports it ideally so should we). If so, seems quite smelly that we need to hack around like this in .NET rather than having a more 1st-class API (is it worth raising the problem more explicitly?).

Otherwise at this point doing something that can't be backported to netfx seems completely fine to me.